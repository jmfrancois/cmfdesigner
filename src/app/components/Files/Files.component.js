import React from 'react';
import PropTypes from 'prop-types';
import { cmfConnect } from '@talend/react-cmf';
import FileAnalytics from '../FileAnalytics';

function Files(props) {
	return (
		<div>
			<h2>{props.title}</h2>
			{props.files.map(file => <FileAnalytics {...file} key={file.path} cwd={props.cwd} />)}
		</div>
	);
}

Files.displayName = 'Files';
Files.propTypes = {
	title: PropTypes.string,
	cwd: PropTypes.string,
	files: PropTypes.arrayOf(PropTypes.object),
};
Files.defaultProps = {
	files: [],
};
export default cmfConnect({})(Files);
