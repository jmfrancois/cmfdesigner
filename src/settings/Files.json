{
  "props": {
    "Files#component": {
      "filesExpression": "getComponentFiles",
      "cwdExpression": "getCWD"
    }
  }
}
