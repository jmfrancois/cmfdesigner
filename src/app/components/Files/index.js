import Files from './Files.component';

export default Files;
