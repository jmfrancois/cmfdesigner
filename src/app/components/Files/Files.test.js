import React from 'react';
import { shallow } from 'enzyme';

import Files from './Files.component';

describe('Files', () => {
	it('should render', () => {
		const wrapper = shallow(
			<Files.WrappedComponent />
		);
		expect(wrapper.getElement()).toMatchSnapshot();
	});
});
