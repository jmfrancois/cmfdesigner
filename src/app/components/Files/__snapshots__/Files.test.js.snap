// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Files should render 1`] = `
<div>
  <h2 />
</div>
`;
